# BDSO Correlation
Compares the source taxonomies of the BDS Ontology with the CxG integrated MoP dataset to evaluate their compatibility

In [48]:
## IMPORT depedencies 

import os
import pandas as pd
import anndata as ad
import numpy as np
import urllib.request as request
import zipfile

## Read h5ad (ann_data)

Please manually download the h5ad file from https://cellxgene.cziscience.com/collections/ae1420fe-6630-46ed-8b3d-cc6056a66467

In [84]:
ann_data = ad.read_h5ad("h5ad-downloads/an-integrated-transcriptomic-cellxgene.h5ad", backed="r")

In [89]:
print(ann_data.obs.shape)
ann_data.obs.head(4)

(406187, 29)


BICCN_cluster_id    QC BICCN_cluster_label  \
SM-D9D8O_S03_E1-50              48.0  1201         L6 CT Grp_1   
SM-DAIH5_S47_E1-50              48.0  3678         L6 CT Grp_1   
SM-DD44L_S43_E1-50               4.0  4495       Lamp5 Slc35d3   
SM-D9E5O_S40_E1-50              49.0  1499         L6 CT Grp_2   

                   BICCN_subclass_label BICCN_class_label cluster_color  \
SM-D9D8O_S03_E1-50                L6 CT     Glutamatergic       #197A6E   
SM-DAIH5_S47_E1-50                L6 CT     Glutamatergic       #197A6E   
SM-DD44L_S43_E1-50                Lamp5         GABAergic       #FFA388   
SM-D9E5O_S40_E1-50                L6 CT     Glutamatergic       #358459   

                     size    temp_class_label BICCN_ontology_term_id  \
SM-D9D8O_S03_E1-50  781.0  GlutamatergicL6 CT            ILX:0770162   
SM-DAIH5_S47_E1-50  781.0  GlutamatergicL6 CT            ILX:0770162   
SM-DD44L_S43_E1-50  194.0      GABAergicLamp5            ILX:0770149   
SM-D9E5O_S40_E1-50   24.0  GlutamatergicL6 CT            ILX:0770162   

                   assay_ontology_term_id  ... donor_id suspension_type  \
SM-D9D8O_S03_E1-50            EFO:0008930  ...   304659            cell   
SM-DAIH5_S47_E1-50            EFO:0008930  ...   319137            cell   
SM-DD44L_S43_E1-50            EFO:0008930  ...   298355            cell   
SM-D9E5O_S40_E1-50            EFO:0008930  ...   306909            cell   

                               cell_type      assay disease      organism  \
SM-D9D8O_S03_E1-50  glutamatergic neuron  Smart-seq  normal  Mus musculus   
SM-DAIH5_S47_E1-50  glutamatergic neuron  Smart-seq  normal  Mus musculus   
SM-DD44L_S43_E1-50      GABAergic neuron  Smart-seq  normal  Mus musculus   
SM-D9E5O_S40_E1-50  glutamatergic neuron  Smart-seq  normal  Mus musculus   

                       sex                tissue self_reported_ethnicity  \
SM-D9D8O_S03_E1-50    male  primary motor cortex                      na   
SM-DAIH5_S47_E1-50    male  primary motor cortex                      na   
SM-DD44L_S43_E1-50    male  primary motor cortex                      na   
SM-D9E5O_S40_E1-50  female  primary motor cortex                      na   

                    development_stage  
SM-D9D8O_S03_E1-50  early adult stage  
SM-DAIH5_S47_E1-50  early adult stage  
SM-DD44L_S43_E1-50  early adult stage  
SM-D9E5O_S40_E1-50  early adult stage  

[4 rows x 29 columns]

## Read cell_to_cell_set_assignments (c2c_data)

Downloads, unzips and loads the cell to cell set assingments data

In [79]:
if not os.path.exists(os.path.join(os.getcwd(), "cell_to_cell_set_assignments_CCN202002013.csv")):
    # Download
    request.urlretrieve("https://github.com/AllenInstitute/MOp_taxonomies_ontology/blob/main/mouseMOp_CCN202002013/cell_to_cell_set_assignments_CCN202002013.zip?raw=true", "cell_to_cell_set_assignments_CCN202002013.zip")

    # Unzip
    with zipfile.ZipFile("cell_to_cell_set_assignments_CCN202002013.zip", 'r') as zip_ref:
        zip_ref.extractall(os.getcwd())
    
# Load to DataFrame
c2c_data = pd.read_csv("cell_to_cell_set_assignments_CCN202002013.csv")

print(c2c_data.shape)
c2c_data.head(5)

(159738, 258)


sample_name  CS202002013_81  \
0  pBICCNsMMrMOpRAiF003d190318_AAACCCAAGGCCTTGC               1   
1  pBICCNsMMrMOpRAiF003d190318_AAACCCAAGGCTAAAT               0   
2  pBICCNsMMrMOpRAiF003d190318_AAACCCATCTGAGCAT               0   
3  pBICCNsMMrMOpRAiF003d190318_AAACGAAAGGGCGAGA               0   
4  pBICCNsMMrMOpRAiF003d190318_AAACGAACAGAGATTA               0   

   CS202002013_1  CS202002013_2  CS202002013_3  CS202002013_4  CS202002013_5  \
0              0              0              0              0              0   
1              0              0              0              0              0   
2              0              0              0              0              0   
3              0              0              0              0              0   
4              0              0              0              0              0   

   CS202002013_6  CS202002013_7  CS202002013_8  ...  CS202002013_248  \
0              0              0              0  ...                0   
1              0              0              0  ...                0   
2              0              0              0  ...                0   
3              0              0              0  ...                0   
4              0              0              0  ...                0   

   CS202002013_249  CS202002013_250  CS202002013_251  CS202002013_252  \
0                0                0                0                1   
1                0                0                0                0   
2                0                0                0                0   
3                0                0                0                0   
4                0                0                0                0   

   CS202002013_253  CS202002013_254  CS202002013_255  CS202002013_256  \
0                0                0                0                0   
1                0                0                0                0   
2                0                0                0                0   
3                0                0                0                0   
4                0                0                0                0   

   CS202002013_257  
0                0  
1                0  
2                0  
3                0  
4                0  

[5 rows x 258 columns]

## Report the differences between c2c_data and ann_data

List all c2cdata sample names that doesn't exist in the anndata

In [91]:
adata_keys = set(ann_data.obs.index.tolist())
cdata_keys = set(c2c_data.sample_name.unique().tolist())
diff = [x for x in cdata_keys if x not in adata_keys]
diff

[]

Seems c2cdata is subset of anndata!

In [108]:
adata_subset = ann_data.obs[ann_data.obs.index.isin(cdata_keys)]
adata_subset.shape

(159738, 29)

## Report the differences between ann_data and c2c_data

Filter ann_data to keep only the matching samples. But ann_data has 406187 rows while c2c_data has only 159738 rows. So check if we need all the ann_data for comparison.

In [93]:
adata_subset.assay.unique()

['10x 3' v3']
Categories (3, object): ['Smart-seq', '10x 3' v2', '10x 3' v3']

In [94]:
adata_subset.suspension_type.unique()

['nucleus']
Categories (2, object): ['cell', 'nucleus']

Seems we are only using assay="10x 3' v3" and suspension_type="nucleus" !

In [103]:
adata_assay = adata.obs[adata.obs["assay"]=="10x 3' v3"]
adata_filtered = target_assay[target_assay["suspension_type"]=="nucleus"]
adata_filtered.shape

(199904, 29)

List all ann_data filtered rows that doesn't exist in the c2cdata

In [109]:
adata_filtered_keys = set(adata_filtered.index.tolist())
diff2 = [x for x in adata_filtered_keys if x not in cdata_keys]
only_anndata = adata_filtered[adata_filtered.index.isin(diff2)]

print(only_anndata.shape)
only_anndata.head(5)

(40166, 29)


BICCN_cluster_id   QC  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02              59.0  1.0   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02              54.0  2.0   
AAACCCACAACCCTAA-1L8TX_181211_01_A02              11.0  3.0   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02              34.0  4.0   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02              30.0  5.0   

                                     BICCN_cluster_label BICCN_subclass_label  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02          L6b Shisa6                  L6b   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02        L6 CT Brinp3                L6 CT   
AAACCCACAACCCTAA-1L8TX_181211_01_A02            Vip Chat                  Vip   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02       L5 IT Rspo1_3                L5 IT   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02           L2/3 IT_2              L2/3 IT   

                                     BICCN_class_label cluster_color    size  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02     Glutamatergic       #2B9880   247.0   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02     Glutamatergic       #338C5E  3970.0   
AAACCCACAACCCTAA-1L8TX_181211_01_A02         GABAergic       #FF00FF   519.0   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02     Glutamatergic       #3CBC45  1838.0   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02     Glutamatergic       #7C8169  2105.0   

                                          temp_class_label  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02      GlutamatergicL6b   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02    GlutamatergicL6 CT   
AAACCCACAACCCTAA-1L8TX_181211_01_A02          GABAergicVip   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02    GlutamatergicL5 IT   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02  GlutamatergicL2/3 IT   

                                     BICCN_ontology_term_id  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02            ILX:0770163   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02            ILX:0770162   
AAACCCACAACCCTAA-1L8TX_181211_01_A02            ILX:0770151   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02            ILX:0770157   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02            ILX:0770156   

                                     assay_ontology_term_id  ... donor_id  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02            EFO:0009922  ...   427311   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02            EFO:0009922  ...   427311   
AAACCCACAACCCTAA-1L8TX_181211_01_A02            EFO:0009922  ...   427311   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02            EFO:0009922  ...   427311   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02            EFO:0009922  ...   427311   

                                     suspension_type             cell_type  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02         nucleus  glutamatergic neuron   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02         nucleus  glutamatergic neuron   
AAACCCACAACCCTAA-1L8TX_181211_01_A02         nucleus      GABAergic neuron   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02         nucleus  glutamatergic neuron   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02         nucleus  glutamatergic neuron   

                                          assay disease      organism     sex  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02  10x 3' v3  normal  Mus musculus  female   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02  10x 3' v3  normal  Mus musculus  female   
AAACCCACAACCCTAA-1L8TX_181211_01_A02  10x 3' v3  normal  Mus musculus  female   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02  10x 3' v3  normal  Mus musculus  female   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02  10x 3' v3  normal  Mus musculus  female   

                                                    tissue  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02  primary motor cortex   
AAACCCAAGTCGAATA-1L8TX_181211_01_A02  primary motor cortex   
AAACCCACAACCCTAA-1L8TX_181211_01_A02  primary motor cortex   
AAACCCAGTATAGCTC-1L8TX_181211_01_A02  primary motor cortex   
AAACCCAGTCAGACTT-1L8TX_181211_01_A02  primary motor cortex   

                                     self_reported_ethnicity  \
AAACCCAAGCTCTTCC-1L8TX_181211_01_A02                      na   
AAACCCAAGTCGAATA-1L